### PR TITLE
chore: setup TypeScript and fix type errors

### PR DESCRIPTION
**Pull request by bajtos**
See https://github.com/space-meridian/roadmap/issues/80

### CODE REVIEW

File: db/index.js [hunk 1]
```diff
@@ -73,7 +76,7 @@ export const getPgPools = async () => {
 }
 
 /**
- * @param {pg.Client} client
+ * @param {pg.Client | pg.Pool} client
```
Comment (bajtos) [inline db/index.js:79]: Unfortunately, in node-pg typings describe Client and Pool in such way that one cannot be assigned to other 😢

File: db/typings.d.ts [hunk 2]
```diff
@@ -1,6 +1,11 @@
 import type { Pool } from 'pg'
 
-export interface pgPools {
+export interface PgPools {
   stats: Pool;
   evaluate: Pool;
 }
+
+export interface EndablePgPools extends PgPools {
+  end(): Promise<unknown>
```
Comment (bajtos) [inline db/typings.d.ts:9]: Our implementation returns `Promise<[void, void]>`. Is it worth changing the implementation so that this line can describe the return value as `void`?

```suggestion
  end(): Promise<void>
```

File: db/index.js [hunk 3]
```diff
@@ -4,6 +4,9 @@ import { dirname, join } from 'node:path'
 import { fileURLToPath } from 'node:url'
 import Postgrator from 'postgrator'
 
+/** @typedef {import('./typings.js').PgPools} PgPools */
```
Comment (bajtos) [inline db/index.js:7]: The convention is to use PascalCase for type names, i.e. start with a capital letter. I renamed `pgPools` type to `PgPools`.

File: db/index.js (hunk 1 above)
Comment (juliangruber) [reply db/index.js:79]: What about we add this type to this repo instead?

File: db/typings.d.ts (hunk 2 above)
Comment (juliangruber) [reply db/typings.d.ts:9]: sgtm! should be a simple change

File: observer/lib/platform-stats.js [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /**
- * @param {import('pg').Client} pgClient
+ * @param {import('pg').ClientBase} pgClient
```
Comment (juliangruber) [inline observer/lib/platform-stats.js:2]: for my understanding, why was this change necessary?

File: stats/bin/spark-stats.js [hunk 5]
```diff
@@ -20,6 +20,6 @@ const logger = {
 const handler = createHandler({ pgPools, logger })
 const server = http.createServer(handler)
 console.log('Starting the http server on host %j port %s', HOST, PORT)
-server.listen(PORT, HOST)
+server.listen(Number(PORT), HOST)
```
Comment (juliangruber) [inline stats/bin/spark-stats.js:23]: what is this cast to number for?

File: stats/test/handler.test.js [hunk 6]
```diff
@@ -38,7 +38,7 @@ describe('HTTP request handler', () => {
     server = http.createServer(handler)
     server.listen()
     await once(server, 'listening')
-    baseUrl = `http://127.0.0.1:${server.address().port}`
+    baseUrl = `http://127.0.0.1:${getPort(server)}`
```
Comment (juliangruber) [inline stats/test/handler.test.js:41]: could we instead do this?
```suggestion
    baseUrl = `http://127.0.0.1:${server.address()!.port}`
```

File: stats/test/platform-routes.test.js [hunk 7]
```diff
@@ -31,7 +31,7 @@ describe('Platform Routes HTTP request handler', () => {
     server = http.createServer(handler)
     server.listen()
     await once(server, 'listening')
-    baseUrl = `http://127.0.0.1:${server.address().port}`
+    baseUrl = `http://127.0.0.1:${getPort(server)}`
```
Comment (juliangruber) [inline stats/test/platform-routes.test.js:34]: as above

**Review by juliangruber — CHANGES_REQUESTED** [review]
massive work 👏

File: observer/lib/platform-stats.js (hunk 4 above)
Comment (bajtos) [reply observer/lib/platform-stats.js:2]: I'll revert this change. Good catch!

I was experimenting with alternatives to the missing `Queryable` interface (see https://github.com/filecoin-station/spark-stats/pull/144#discussion_r1637976607) and forgot to revert this place.

File: stats/bin/spark-stats.js (hunk 5 above)
Comment (bajtos) [reply stats/bin/spark-stats.js:23]: The way how TypeScript describes the `listen` method, it accepts only numbers as the port. When we use `PORT`, which can be a string, TypeScript complains.

```js
server.listen('8080', 'localhost')
```

File: stats/test/handler.test.js (hunk 6 above)
Comment (bajtos) [reply stats/test/handler.test.js:41]: Unfortunately not. 

`server.address()` could be either a string (when listening on a named pipe(?)) or an object with a `port` property (when listening over TCP/IP).

We know that `address` will always be an object, but TypeScript cannot infer that.

It's annoying.

File: db/index.js (hunk 1 above)
Comment (bajtos) [reply db/index.js:79]: Hang on! Maybe this will work - let me test it:
```ts
export type Queryable = Pick<Pool, 'query'>
```

Comment (bajtos) [reply db/index.js]: Works like a charm! See [0324c01](https://github.com/filecoin-station/spark-stats/pull/144/commits/0324c010c4f6817313c12f42b56a3b5e44eda1e9)

File: db/typings.d.ts (hunk 2 above)
Comment (bajtos) [reply db/typings.d.ts]: Done in [4741c76](https://github.com/filecoin-station/spark-stats/pull/144/commits/4741c7660a1c230d45d1e5602e862c141511d427)

**Review by juliangruber — APPROVED** [review]
😍
